lib/proxy/sd_notify.rb

```diff
@@ -0,0 +1,27 @@
+require 'socket'
+
+# Implementation of libsystemd's sd_notify API, sends current state via socket
+module Proxy
+  class SdNotify
+    def active?
+      !ENV['NOTIFY_SOCKET'].nil?
+    end
+
+    def notify(message)
+      socket.sendmsg(message.chomp + "\n") # ensure trailing \n
```

Comment:
You aren't closing the socket when you are done with it. Or is this intentional so other changes in state (restarts, for example) can be passed to systemd?